src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -300,11 +377,33 @@ private synchronized void freeResources(List<String> unlockResourceNames, @Nulla
 						// No more contexts, unlock resource
 						resource.unqueue();
 						resource.setBuild(null);
+						// If the resource is on the stratos server delete the reservation objective
+						if (fromName(resource.getName(), stratosResources) !=null){
+							LOGGER.log(Level.FINE, "Freeing " + resource.getName() + " from " + getStratosURL());
+							String labUID = resource.getUid();
+							List<Objective> reservationObjectives = controllerAPI.objectives().getByType("SUDS/1.0/reservation");
+							for(Objective currentReservation : reservationObjectives){
+								String reservationUID = getUidFromSelfLink(currentReservation.getSelf());
+								List<String> governors = currentReservation.getGovernors();
+								for (String currentGov : governors){
+									String currentGovUid = getUidFromSelfLink(currentGov);
+									if(currentGovUid.equals(labUID)){
+										LOGGER.log(Level.FINE, "Deleting reservation: " + reservationUID);
+										controllerAPI.objectives().destroy(reservationUID);
+									}
+								}
+							}
+						}
 					}
 				}
 			}
 		}
 	}
+	
```

Comment:
It doesn't make sense that this should be a function of LockableResourceManager since it is really a function to do with objectives. It should probably be added to the Objective class?